src/server/scripts/Spells/spell_shaman.cpp

```diff
@@ -82,6 +84,85 @@ enum MiscSpells
     SPELL_PET_NETHERWINDS_FATIGUED              = 160455
 };
 
+// 108281 - Ancestral Guidance
+class spell_sha_ancestral_guidance : public SpellScriptLoader
+{
+    public:
+        spell_sha_ancestral_guidance() : SpellScriptLoader("spell_sha_ancestral_guidance") { }
+
+        class spell_sha_ancestral_guidance_AuraScript : public AuraScript
+        {
+            PrepareAuraScript(spell_sha_ancestral_guidance_AuraScript);
+
+            bool Validate(SpellInfo const* /*spellInfo*/) override
+            {
+                if (!sSpellMgr->GetSpellInfo(SPELL_SHAMAN_ANCESTRAL_GUIDANCE))
+                    return false;
+                if (!sSpellMgr->GetSpellInfo(SPELL_SHAMAN_ANCESTRAL_GUIDANCE_HEAL))
+                    return false;
+                return true;
+            }
+
+            void HandleEffectProc(AuraEffect const* aurEff, ProcEventInfo& eventInfo)
+            {
+                PreventDefaultAction();
+                // No heal from this talent
+                if (eventInfo.GetHealInfo()->GetSpellInfo()->Id == SPELL_SHAMAN_ANCESTRAL_GUIDANCE_HEAL)
+                    return;
+                int32 bp0 = CalculatePct(int32(eventInfo.GetDamageInfo()->GetDamage() + eventInfo.GetHealInfo()->GetHeal()), aurEff->GetAmount());
```

Comment:
this won't work correctly with the current "hacked" implementation of eventinfo, `eventInfo.GetDamageInfo()->GetDamage()` and `eventInfo.GetHealInfo()->GetHeal()` holds the same data